target/snitch_cluster/cfg/snax-streamer-gemmX-xdma.hjson

```diff
@@ -136,9 +136,9 @@
             writer_buffer: 8, 
             reader_agu_dimension: 3, 
             writer_agu_dimension: 3, 
-            has_transposer: true, 
-            has_maxpool: true, 
-            has_memset: true
+            HasTransposer: 3, 
+            HasMemset: 1, 
+            HasMaxPool: 2
```

Comment:
What are the numbers for? Number of instances or position in the extension?

Reply:
Yes. When you set it to <=0, then the extension is disabled. Setting it to larger than 0 to enable. The larger the number it is, it will be more to the end of the extension chain. You can even write like: 

HasTransposer: 1
HasUserExtension 2
HasTransposer: 3

To instantiate transposer twice, one in front user extension and another at the back of extension. 

Furthermore, now you don't need to modify anything related to generator. (bender.yml, Makefile, Chisel generator, Python etc. ) When adding a new custom extension to xdma, what you only need to do is add the actual implementation and add the item (like listed here) to the configuration. Chisel will handle everything for you.

This is all made possible by incorporating the Meta programmability of Scala, which Scala does not only generate SystemVerilog, but generate Scala and execute it.